test/aqua/test_phase_estimator.py

```diff
@@ -0,0 +1,200 @@
+        a1 = 0.5
+        a2 = 1.0
+        hamiltonian = (a1 * X) + (a2 * Z)
+        state_preparation = H.to_circuit()
+        result = self.hamiltonian_pe(hamiltonian, state_preparation)
+        phase_dict = result.filter_phases(0.162, as_float=True)
+        phases = list(phase_dict.keys())
+        self.assertAlmostEqual(phases[0], 1.119, delta=0.001)
+        self.assertAlmostEqual(phases[1], -1.119, delta=0.001)
+
+    # pylint: disable=invalid-name
+    def _hamiltonian_1(self):
+        a1 = 0.5
+        a2 = 1.0
+        a3 = 1.0
+        hamiltonian = (a1 * X) + (a2 * Y) + (a3 * Z)
+        return hamiltonian
+
+    def test_pauli_sum_2(self):
+        """Two eigenvalues from Pauli sum with X, Y, Z"""
+        hamiltonian = self._hamiltonian_1()
+        state_preparation = None
+        result = self.hamiltonian_pe(hamiltonian, state_preparation)
+        phase_dict = result.filter_phases(0.1, as_float=True)
+        phases = list(phase_dict.keys())
+        self.assertAlmostEqual(phases[0], 1.5, delta=0.001)
+        self.assertAlmostEqual(phases[1], -1.5, delta=0.001)
+
+    def _setup_from_bound(self, evolution):
+        hamiltonian = self._hamiltonian_1()
+        state_preparation = None
+        bound = 1.2 * sum([abs(hamiltonian.coeff * pauli.coeff) for pauli in hamiltonian])
+        backend = qiskit.Aer.get_backend('qasm_simulator')
+        qi = qiskit.aqua.QuantumInstance(backend=backend, shots=100000)
+        phase_est = HamiltonianPE(num_evaluation_qubits=8,
+                                  hamiltonian=hamiltonian,
+                                  bound=bound,
+                                  quantum_instance=qi,
+                                  state_preparation=state_preparation,
+                                  evolution=evolution)
+        result = phase_est.run()
+        return result
+
+    def test_from_bound(self):
+        """HamiltonianPE with bound"""
+        result = self._setup_from_bound(TempPauliEvolve())
+        phases = result.filter_phases()
+        self.assertEqual(len(phases), 2)
+        self.assertEqual(list(phases.keys()), [1.5, -1.5])
+        phases = result.filter_phases(scaled=False)
+        self.assertEqual(list(phases.keys()), [0.25, 0.75])
+        self.assertEqual(result.single_phase(), -1.5)
+        self.assertEqual(result.single_phase(scaled=False), 0.75)
+
+    def test_trotter_from_bound(self):
+        """HamiltonianPE with bound and Trotterization"""
+        result = self._setup_from_bound(PauliTrotterEvolution(trotter_mode='suzuki', reps=3))
+        phase_dict = result.filter_phases(0.1)
+        phases = list(phase_dict.keys())
+        self.assertEqual(len(phases), 2)
+        self.assertAlmostEqual(phases[0], 1.476, delta=0.001)
+        self.assertAlmostEqual(phases[1], -1.476, delta=0.001)
+
+
+class TestPhaseEstimator(QiskitAquaTestCase):
+    """Evolution tests."""
+
+    # pylint: disable=invalid-name
```

Comment:
you don't have to put this if you rename `qi` to `q_instance` or `quantum_instance` 🙂  I think less pylint ignores is generally better if they can be worked around easily